### ISSUE TITLE
Memory leak from ofxCvContourFinder findContours

### ISSUE THREAD

**Issue by markwheeler** [openframeworks/openFrameworks#4460]
I'm seeing a memory leak when using ofxCvContourFinder's findContours()

The below sample code causes this leak when put in the update method of an otherwise empty project:

    ofxCvGrayscaleImage testCvImg;
    testCvImg.allocate( 640, 480 );
    
    ofxCvContourFinder contourFinder;
    contourFinder.findContours( testCvImg, 20, 200, 5, false, false );

I've tested this with 0.9.0RC1 and the latest master on 2015-10-21 by running Xcode's 'Profile' and monitoring the allocations – I see a steady increase in persistent malloc 64s. I'm on OSX 10.10

With oF 0.8.4 I don't see the leak.

Let me know if you'd like me to test something else or if I missed something :)

**Comment by ofTheo** [comment]
if you hook it up to a key being pressed I would be curious to see if you see the memory being released when the contourFinder.findContours is not running. 

ie: wrap that code in a 

```
if( ofGetKeyPressed('t') ){

}
```

**Comment by arturoc** [comment]
i'm running the same code in ubuntu whith the same version of openCv and the memory usage is totally stable for me. i've also analized it with valgrind and i don't see any memory leaks.

how much memory is it using in total and over how much does it increase over time for you? some times in some platforms applications appear to be leaking memory until they stabilize after a while so try also to keep the application running for at least 10 minutes and see if it still increases.

also the leaks tool in xcode instruments should tell you if there's a leak and in which part of the code is happening

**Comment by markwheeler** [comment]
Thanks for looking into this!

@ofTheo ah interesting, I just tried this and it does indeed release eventually. About 5-10 seconds after i let go of the key the mem usage drops back down.

@arturoc I just tried running the above code for 25mins and it was continuing to grow in mem usage. Here are a couple of screenshots:
http://imgur.com/a/bquKg
The first is soon after the app starts running and it's using just under 9MB total, the second is after 25 mins and it's at 21.5MB and still increasing. 'Malloc 64 bytes' has gone from 8900 persistent to 218000 persistent.

I guess the first test proves that it's not actually leaking, but rather it's hanging onto memory?
